core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -5758,14 +5758,21 @@ private void translateAgg(SqlCall call, @Nullable SqlNode filter,
       if (orderList == null || orderList.size() == 0) {
         collation = RelCollations.EMPTY;
       } else {
-        collation = RelCollations.of(
-            orderList.stream()
-                .map(order ->
-                    bb.convertSortExpression(order,
-                        RelFieldCollation.Direction.ASCENDING,
-                        RelFieldCollation.NullDirection.UNSPECIFIED,
-                        this::sortToFieldCollation))
-                .collect(Collectors.toList()));
+        try {
+          // switch out of agg mode
+          bb.agg = null;
```

Comment:
My question is why not keep the reference of bb.agg and then in the finally block, assign that reference back? (or is the bb.agg is always equal to this class?)

Reply:
It must be equal to this class. If you scroll at the top of this function, you'll see that. We also do something similar earlier in this function as well.

Reply:
See: https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java#L5585
and:
https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java#L5700

Reply:
Thanks for clarifying. 

I can see now this is already an existing practice.